### PR TITLE
Merge Bomb with MrBomb

### PR DESCRIPTION
**Pull request by MatusGuy**
This PR internally rewrites MrBomb to be more like GoldBomb. It doesn't add any considerable features.

It also removes lots of code duplication. The Bomb class is now obsolete.

As a result, the death script stays consistent between MrBomb and GoldBomb.

This PR also fixes a bug with gold bomb, in which getting out of the safe distance and going back in would freeze gold bomb in the same sprite action until another state change.

### CODE REVIEW

**Review by tylerandari13 — APPROVED** [review]
Just what I needed. Works great.

**Review by Frostwithasideofsalt — APPROVED** [review]
yeah this really does work yup mhm

**Review by weluvgoatz — APPROVED** [review]

File: src/badguy/mrbomb.cpp [hunk 1]
```diff
@@ -1,5 +1,7 @@
-//  SuperTux
+//  SuperTux BadGuy MrBomb
```
Comment (Vankata453) [inline src/badguy/mrbomb.cpp:1]: I noticed we don't modify this part of the copyright header anymore. Should we modify it here?

File: src/badguy/goldbomb.hpp [hunk 2]
```diff
@@ -19,66 +19,42 @@
...
-  virtual void freeze() override;
-  virtual bool is_freezable() const override;
-
-  virtual void kill_fall() override;
-  virtual void ignite() override;
   static std::string class_name() { return "goldbomb"; }
   virtual std::string get_class_name() const override { return class_name(); }
   static std::string display_name() { return _("Gold Bomb"); }
   virtual std::string get_display_name() const override { return display_name(); }
-  virtual GameObjectClasses get_class_types() const override { return WalkingBadguy::get_class_types().add(typeid(GoldBomb)); }
```
Comment (Vankata453) [inline src/badguy/goldbomb.hpp:52]: Why was this removed?

File: src/badguy/goldbomb.hpp [hunk 3]
```diff
@@ -19,66 +19,42 @@
...
+  void recover();
+  void normalize();
 
 private:
-  enum Ticking_State {
-    STATE_NORMAL,
-    STATE_TICKING,
-    STATE_REALIZING,
+  enum State : uint8_t {
+    STATE_REALIZING = 2,
```
Comment (Vankata453) [inline src/badguy/goldbomb.hpp:50]: In the `MrBomb::State` enumerators, add one at the end called `STATE_LAST`. Use it here to set `STATE_REALIZING`, instead of hardcoding the value in this file (may prevent future bugs when updating `MrBomb::State`).

**Review by Vankata453 — COMMENTED** [review]
Mr. Bomb and Gold Bomb seem to work fine. I'm not sure what the mentioned issue was, so I trust you that it's been fixed.

Left some code-related comments.

File: src/badguy/goldbomb.hpp (hunk 2 above)
Comment (MatusGuy) [reply src/badguy/goldbomb.hpp:52]: Must've been the automatic merge

**Comment by MatusGuy** [comment]
@weluvgoatz I accidentally requested your review.

File: src/badguy/goldbomb.hpp [hunk 4]
```diff
@@ -19,66 +19,43 @@
...
 
 private:
-  enum Ticking_State {
-    STATE_NORMAL,
-    STATE_TICKING,
-    STATE_REALIZING,
-    STATE_FLEEING,
-    STATE_CORNERED
+  enum State : uint8_t {
+    GB_STATE_REALIZING = 2,
```
Comment (Vankata453) [inline src/badguy/goldbomb.hpp:51]: This should be `MB_STATE_COUNT`:

```suggestion
    GB_STATE_REALIZING = MB_STATE_COUNT,
```
